crates/pallet-domains/src/lib.rs

```diff
@@ -1522,6 +1526,25 @@ impl<T: Config> Pallet<T> {
                 >(bad_receipt, storage_proof)
                 .map_err(FraudProofError::InvalidTotalRewardsFraudProof)?;
             }
+            FraudProof::InvalidDomainBlockHash(InvalidDomainBlockHashProof {
+                digest_storage_proof,
+                ..
+            }) => {
+                let parent_receipt =
+                    DomainBlocks::<T>::get(bad_receipt.parent_domain_block_receipt_hash)
+                        .ok_or(FraudProofError::BadReceiptNotFound)?
```

Comment:
We should introduce a new error instead of using `BadReceiptNotFound`, because the parent ER of the targetted bad ER is not a bad ER, if it is a bad ER then the honest operator should submit a fraud proof that targets it instead.